pib_api/flask/controller/pose_controller.py

```diff
@@ -0,0 +1,61 @@
+from service import pose_service
+from schema.pose_schema import (
+    poses_schema,
+    pose_schema,
+    pose_schema_motor_positions_only,
+    create_pose_schema,
+    pose_schema_name_only,
+    pose_schema_without_motor_positions,
+)
+from app.app import db
+from flask import abort, jsonify, request, Blueprint
+
+
+bp = Blueprint("pose_controller", __name__)
+
+
+@bp.route("", methods=["POST"])
+def create_pose():
+    pose_dto = create_pose_schema.load(request.json)
+    pose = pose_service.create_pose(pose_dto)
+    db.session.commit()
```

Comment:
Why commit in the controller instead of the service?
Shouldn't the service layer be responsible for handling the business logic and the controller layer for validating data and delegating work? Completing the DB transaction in the controller goes against this separation.

Reply:
I don't think the service-layer is the correct place to perform the commit. If f.e. the controller calls multiple service-methods and one of the calls fails, we will be left with partial changes, which we do not want. Our transactions should be request-scoped. I solved this issue now, by moving the commits into a function in the 'app.py' that has a '@after_request' decorator (i.e. it fires after a request was handled by the controller). The function checks, if the response-code indicates a success and based on this information, it either rolls-back or commits.

Reply:
In those cases wouldn't a method inside the service that calls the other submethods in the service and reverses the transaction upon failure also work and clearly indicate where a transaction starts and ends?
With the current solution after every request the DB is committed, even if the endpoint does not start a transaction. According to the SQLAlchemy documentation, this should raise a `InvalidRequestError`. If that is not the case, I am also fine if we don't change anything as it goes beyond the scope of this ticket.

Reply:
> In those cases wouldn't a method inside the service that calls the other submethods in the service and reverses the transaction upon failure also work

But wouldn't that method need to know what changes the other methods have done? This sounds rather convoluted and error-prone.

> and clearly indicate where a transaction starts and ends?

I don't see how this makes anything clearer. I think the current approach makes this already pretty clear and it is consistent with sqlalchemy's behavior of creating request-scoped sessions.

> With the current solution after every request the DB is committed, even if the endpoint does not start a transaction. According to the SQLAlchemy documentation, this should raise a InvalidRequestError

Where did you get that information from? It says [here](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.commit) that if there is no transaction in place, the `session.commit()` message will create a 'internal-only logical transaction'. I also tested it by explicitly throwing an error in a method that does not interact with the db and everything went fine.